docs/architecture/adr-015-namespace-id-size.md

```diff
@@ -0,0 +1,58 @@
+# ADR 15: Namespace ID Size
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/2/17: initial draft
+
+## Context
+
+Namespace ID is currently an 8 byte slice. 8 bytes provides a maximum of 2^64 possible namespace IDs. In practice some namespace IDs are reserved for protocol use so the number of namespace IDs available to users is 2^56 - 2. Modifying the size of a namespace ID post-launch is a breaking change and has implications for the NMT, share encoding, etc. so we'd like to carefully consider the size of the namespace ID pre-launch.
+
+Desirable criteria:
+
+1. Ability to randomly generate a namespace that hasn't been used before
```

Comment:
```suggestion
1. Ability to randomly generate a namespace that hasn't been used before (with large enough probability)
```

Other cirteria:

- Potentially large enough for all rollups that will (ever) exist 
- use a truncated secure hash function (like sha256) for generating the namespace 

The 2nd though is basically the same as "Ability to randomly generate a namespace that hasn't been used before".

Reply:
> Potentially large enough for all rollups that will (ever) exist

Will add.

> The 2nd though is basically the same as "Ability to randomly generate a namespace that hasn't been used before".

Agreed so there is a bullet point for it in the Notes section but I didn't include it as an explicit desirable criteria.